contract/market/src/impl_helper.rs

```diff
@@ -378,4 +378,40 @@ impl Contract {
             borrow_position.record_collateral_asset_deposit(amount);
         }
     }
+
+    #[private]
+    pub fn withdraw_static_yield_01_finalize(
+        &mut self,
+        account_id: AccountId,
+        borrow_asset_amount: BorrowAssetAmount,
+        collateral_asset_amount: CollateralAssetAmount,
+    ) {
+        let mut static_yield = self.static_yield.get(&account_id).unwrap_or_else(|| {
+            env::panic_str("Invariant violation: static yield entry must exist during callback")
+        });
+        let mut i = 0;
+
+        if !borrow_asset_amount.is_zero() {
+            if matches!(env::promise_result(i), PromiseResult::Failed) {
+                static_yield
+                    .borrow_asset
+                    .join(borrow_asset_amount)
+                    .unwrap_or_else(|| {
+                        env::panic_str("Borrow asset static yield returned overflows")
+                    });
+            }
+            i += 1;
+        }
```

Comment:
This doesn't have to be `mut`. Something like the following could be nicer:
```suggestion
        let i = if borrow_asset_amount.is_zero() {
            0
        } else {
            if matches!(env::promise_result(i), PromiseResult::Failed) {
                static_yield
                    .borrow_asset
                    .join(borrow_asset_amount)
                    .unwrap_or_else(|| {
                        env::panic_str("Borrow asset static yield returned overflows")
                    });
            }
            1
        };
```